### PR TITLE
Implement priority render.

### PR DESCRIPTION
**Pull request by patrykkolodziej**

### CODE REVIEW

File: zgui.hh [hunk 1]
```diff
@@ -176,10 +176,28 @@ namespace zgui {
 		zgui_dummy
 	};
```
Comment (danielkrupinski) [inline zgui.hh]: Maybe `enum class` would fit better here?

File: zgui.hh [hunk 2]
```diff
@@ -176,10 +176,28 @@ namespace zgui {
...
 
+	enum zgui_render_type 
+	{
+		zgui_draw_line = 1,
+		zgui_draw_rect,
+		zgui_draw_filled_rect,
+		zgui_draw_text
+	};
+
+	struct zgui_control_render_t
```
Comment (danielkrupinski) [inline zgui.hh]: Typo

File: zgui.hh [hunk 3]
```diff
@@ -176,10 +176,28 @@ namespace zgui {
...
+	{
+		vec2 draw_posistion;
+		zgui_render_type render_type;
```
Comment (danielkrupinski) [inline zgui.hh]: Non-owning character string, may be dangling pointer.

File: zgui.hh [hunk 4]
```diff
@@ -176,10 +176,28 @@ namespace zgui {
...
+	{
+		zgui_draw_line = 1,
+		zgui_draw_rect,
+		zgui_draw_filled_rect,
+		zgui_draw_text
+	};
+
+	struct zgui_control_render_t
+	{
+		vec2 draw_posistion;
```
Comment (danielkrupinski) [inline zgui.hh]: Typo

File: zgui.hh [hunk 5]
```diff
@@ -176,10 +176,28 @@ namespace zgui {
...
+		zgui_render_type render_type;
+		color color;
+		const char* text;
```
Comment (danielkrupinski) [inline zgui.hh]: Non-owning character string, possible dangling pointer.

**Comment by zxvnme** [comment]
Thanks for contributing.

**Pull request merged**